docs: Add README.md file

Ksh1tij777 opened the pull request:
Hi, Added a README.md file for the project

Closes #1

This PR adds a detailed README.md to the root of the repository. It includes:

1)Project overview and purpose

2)Tech stack and prerequisites

3)Docker-based setup instructions

4)guidelines

5)License

This improves project onboarding and documentation.

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Documentation**
  * Added a comprehensive README file with an overview of the platform, key features, tech stack details, setup instructions, and local development guidance.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Ksh1tij777 committed:
added README.md

ItamarShalev reviewed (changes requested):
Well done and thanks for your first patch !

Ksh1tij777 commented on README.md:
```diff
@@ -0,0 +1,59 @@
+# 📚 summaries_website
```
ok will make the neccessary changes

Ksh1tij777 commented on README.md:
```diff
@@ -0,0 +1,59 @@
+# 3. To stop
+docker-compose down
+
+---
+
+🤝 Contributing
+We welcome contributions from students and developers!
+Feel free to fork, create a branch, and open a PR.
+
+---
+
+📄 License
+This project is open-source under the MIT License.
+
+---
```
alright thanks a ton for your feedback ! will make the neccessary changes..

ItamarShalev commented on README.md:
```diff
@@ -0,0 +1,59 @@
+
+---
+
+## 🚀 Features
+
+- Browse and share well-organized academic summaries  
+- Built by students — open-source and community-driven  
+- Modern frontend + backend architecture  
+- Easy to run with Docker and Docker Compose  
+
+---
+
+## 🛠 Tech Stack
+
+- **Frontend**: React + Vite + pnpm  
+- **Backend**: FastAPI + uvicorn  
+- **Package Managers**: `pnpm`, `uv`  
+- **Containerization**: Docker & Docker Compose  
+
+---
+
+## ✅ Prerequisites
```
Amazing, Also, add a section what needed to be installed (Docker for example)

and link to the readme.md of the backend and the fronted 

Champion !

ItamarShalev commented:
btw
the common commit title for it it's:
docs: Add README.md file 


We always use "command" as the title, like:
Add, Remove, Create, etc

and never use:
Added, removed, created etc

Ksh1tij777 committed:
docs: Add README.md file

Signed-off-by: Kshitij Verma <vermakshitij747@gmail.com>

Ksh1tij777 commented:
Hi! I've made the requested changes:
- Updated the commit message to match the template
- Added the missing sign-off
-Have removed the contributing and license part
-Made the Language more natural
-Added a section about what needed to be installed and linked to the readme.md of the backend and the fronted
-Changed the Docker code as my Docker commands use the legacy docker-compose binary. The recommended CLI since Docker 20.10 is docker compose … (space, no dash).
-Added a well defined project structure
-Added notes on Docker best practices (e.g., .dockerignore, base image recommendations, layer caching)

Thanks a ton for supporting and giving me appropriate feedback. Let me know if anything else has to be changed or added !!

ItamarShalev reviewed (changes requested):
Small fixes left, well done !

Ksh1tij777 commented on README.md:
```diff
@@ -0,0 +1,83 @@
+
+-A simple way to browse and share course summaries with others
+
+-Built by students from different universities,open and community-driven
```
Alright sure,  will make the changes

Ksh1tij777 commented on README.md:
````diff
@@ -0,0 +1,83 @@
+
+---
+
+## 📂 Project structure
+
+```text
+.
+├── backend/
+│   ├── Dockerfile
+│   └── …
+├── frontend/
+│   ├── Dockerfile
+│   └── …
+├── docker-compose.yml
+└── README.md
````
I checked so there was a docker folder under frontend and backend and it contained te dockerfile. No worries I will remove it !

Ksh1tij777 committed:
docs: Add README.md file

Signed-off-by: Kshitij Verma <vermakshitij747@gmail.com>

ItamarShalev commented:
Thanks for your first patch !
Well done !

Ksh1tij777 commented:
Thanks a ton for the opportunity, really appreciate it !
